From 7073235cef85b6e3078b6fb2bbbe48e0ea5d371c Mon Sep 17 00:00:00 2001
From: "Carsten Haitzler (Rasterman)" <raster@rasterman.com>
Date: Sat, 22 May 2021 14:23:47 +0100
Subject: [PATCH 10/13] alpm - fix bug waiting to happen with cast from ptr to
 boolean

if pointer values are just right this could be false when it shouldnt
be. convert to x ? a : b style to handle this.
---
 backends/alpm/pk-alpm-sync.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backends/alpm/pk-alpm-sync.c b/backends/alpm/pk-alpm-sync.c
index 747b263f5..abc6fb1d5 100644
--- a/backends/alpm/pk-alpm-sync.c
+++ b/backends/alpm/pk-alpm-sync.c
@@ -201,7 +201,7 @@ pk_backend_sync_thread (PkBackendJob* job, GVariant* params, gpointer p)
 		alpm_flags |= ALPM_TRANS_FLAG_DOWNLOADONLY;
 
 	if (pk_alpm_transaction_initialize (job, alpm_flags, NULL, &error) &&
-	    pk_alpm_transaction_sync_targets (job, package_ids, (gboolean)p, &error) &&
+	    pk_alpm_transaction_sync_targets (job, package_ids, p ? TRUE : FALSE, &error) &&
 	    pk_alpm_transaction_simulate (job, &error)) {
 
 		if (pk_bitfield_contain (flags, PK_TRANSACTION_FLAG_ENUM_SIMULATE)) { /* simulation */
-- 
2.31.1

